prob-dep-types/inf_order_probs.metta

```diff
@@ -0,0 +1,92 @@
+(: Distribution (-> Type Type))
+(: Union (-> Type Type Type))
+(= (Union $t1 $t2) $t1)
+(= (Union $t1 $t2) $t2)
+(= (random $a $b) $a)
+(= (random $a $b) $b)
+(: sample (-> (Distribution $t1) $t1))
+(: thunk (-> Atom (Distribution $t1)))
+(= (sample (thunk $a)) $a)
```

Comment:
This seems to be identical to what is in `prob_dep_types.metta`.  I suppose it could be moved to its own file and import it in both places.